src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -609,6 +609,11 @@ class spell_sha_fire_nova : public SpellScriptLoader
                     return SPELL_FAILED_CUSTOM_ERROR;
                 }
 
+                float distance = 30.0f;
+                Creature* totem = GetCaster()->GetMap()->GetCreature(GetCaster()->m_SummonSlot[1]);
```

Comment:
Use object accessor, as @ariel- suggested, but you should always opt for non-hardcoded values. Like check if the original spell has rangeindex set, evaluate that, use that value. Same summonslot. Get the spelleffect that summons the totem, miscvalueb gives the summonproperties entry, look it up in the store, use the "slot" field to get totem from m_SummonSlot.
These are all optional, since 3.3.5 is not going to change anytime soon, but it's a good habit to always use dbc values, so if blizzard changes them, you won't have to bother with changing these scripts